src/App.tsx

```diff
@@ -14,11 +14,132 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const [isShown, setIsShown] = useState(false);
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [length, setLength] = useState('1');
+
+  const reverse = () => {
+    setGoods([...goods].reverse().filter(good => good.length >= +length));
+  };
+
+  const sort = () => {
+    setGoods([...goods]
+      .sort((a, b) => a.localeCompare(b))
+      .filter(good => good.length >= +length));
+  };
+
+  const reset = () => {
+    setGoods(goodsFromServer);
+    setLength('1');
+  };
+
+  const sortByLength = () => {
+    setGoods([...goods].sort((a, b) => b.length - a.length));
+  };
+
+  return (
+    <div className="App">
+      {
+        !isShown
+          ? (
+            <button
+              type="button"
+              className="button"
+              onClick={() => setIsShown(true)}
+            >
+              Start
+            </button>
```

Comment:
better do not use such small check in return and do not complicate structure of `return`
```suggestion
  const sortByLength = () => {
    setGoods([...goods].sort((a, b) => b.length - a.length));
  };
  
  if (!isShown) {
    return (your button here)
  }

  return (
    <div className="App">
      {
        !isShown
          ? (
            <button
              type="button"
              className="button"
              onClick={() => setIsShown(true)}
            >
              Start
            </button>
```
You can do like this and remove conditional rendering here